src/backend/aspen/test_infra/models/pathogen_repo_config.py

```diff
@@ -1,35 +1,65 @@
+import random
+from typing import Dict, Optional, Tuple
+
 from aspen.database.models import Pathogen, PathogenRepoConfig, PublicRepository
 from aspen.database.models.usergroup import generate_random_id
 from aspen.test_infra.models.pathogen import random_pathogen_factory
+from aspen.util.split import SplitClient
 
 
-def setup_gisaid_and_genbank_repo_configs(
+def pathogen_repo_config_factory(prefix, pathogen, public_repository):
+    pathogen_repo_config = PathogenRepoConfig(
+        prefix=prefix, pathogen=pathogen, public_repository=public_repository
+    )
+
+    return pathogen_repo_config
+
+
+def setup_random_repo_configs(
     async_session,
-    pathogen=None,
-    gisaid_prefix=generate_random_id(5),
-    genbank_prefix=generate_random_id(5),
-) -> Pathogen:
-
-    repo_info = {
-        "GISAID": {"prefix": gisaid_prefix},
-        "GenBank": {"prefix": genbank_prefix},
-    }
-    if pathogen is None:
+    pathogen: Optional[Pathogen] = None,
+    repos: Dict[str, str] = None,
+    split_client: Optional[SplitClient] = None,
+    default_repo: Optional[str] = None,
+) -> Tuple[Pathogen, PathogenRepoConfig]:
+    if not repos:
+        repos = {
+            generate_random_id(10): generate_random_id(5),
+            generate_random_id(10): generate_random_id(5),
+        }
+    if not pathogen:
         pathogen = random_pathogen_factory()
         async_session.add(pathogen)
-    for name, config in repo_info.items():
+    repo_configs = {}
+    for name, prefix in repos.items():
         public_repository = PublicRepository(name=name)
         async_session.add(public_repository)
         pathogen_repo_config = pathogen_repo_config_factory(
-            config["prefix"], pathogen, public_repository
+            prefix, pathogen, public_repository
         )
+        repo_configs[public_repository.name] = pathogen_repo_config
         async_session.add(pathogen_repo_config)
-    return pathogen
+    # Randomly choose a default pathogen repo config
+    default_config = random.choice(list(repo_configs.values()))
+    if default_repo:
+        default_config = repo_configs[default_repo]
+    if split_client:
+        split_client.get_pathogen_treatment.return_value = (
+            default_config.public_repository.name
+        )
+    return pathogen, default_config
 
 
-def pathogen_repo_config_factory(prefix, pathogen, public_repository):
-    pathogen_repo_config = PathogenRepoConfig(
-        prefix=prefix, pathogen=pathogen, public_repository=public_repository
+def setup_gisaid_and_genbank_repo_configs(
```

Comment:
For the few endpoints that have specific `gisaid` vs `genbank` switches (like the metadata download), make it easy to create those public repositories instead of random ones.